### PR TITLE
gomacro: init at 2.7

### PR DESCRIPTION
**Pull request by kf5grd**
###### Motivation for this change
Adds gomacro

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -8417,6 +8417,12 @@
     githubId = 997855;
     name = "Narazaki Shuji";
   };
+  shofius = {
```
Comment (fabaff) [inline maintainers/maintainer-list.nix:8420]: Could you please do the adding to `maintainer-list.nix ` in a separate commit?

File: pkgs/development/tools/gomacro/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
+  pname = "gomacro";
+  version = "2.7";
+
+  src = fetchFromGitHub {
+    owner = "cosmos72";
+    repo = "gomacro";
```
Comment (fabaff) [inline pkgs/development/tools/gomacro/default.nix:9]: Common practice

```suggestion
    repo = pname;
```

File: pkgs/development/tools/gomacro/default.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
+  pname = "gomacro";
+  version = "2.7";
```
Comment (fabaff) [inline pkgs/development/tools/gomacro/default.nix:5]: Technically this package is "post 2.7" thus I would mention it somehow. In this case I would to do something like this:

```suggestion
  version = "20210131-${lib.strings.substring 0 7 rev}";
  rev = "b4c3ab9b218fd12f22759354f4f3e37635828d1f";
```

Using a date make it easy for a human to get if the package is up-to-date or not. Adding a part of the hash allows one to go directly to the state in the source which was used. 

There are other approaches of course, e. g. using `unstable` and a like.

File: pkgs/development/tools/gomacro/default.nix [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+  src = fetchFromGitHub {
+    owner = "cosmos72";
+    repo = "gomacro";
+    rev = "b4c3ab9b218fd12f22759354f4f3e37635828d1f";
```
Comment (fabaff) [inline pkgs/development/tools/gomacro/default.nix:10]: If the mentioned change from above is used. 

```suggestion
    inherit rev;
```

**Review by fabaff — CHANGES_REQUESTED** [review]

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 111405` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gomacro</li>
  </ul>
</details>

File: pkgs/development/tools/gomacro/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/development/tools/gomacro/default.nix:9]: Actually I don't think this should be done everywhere. It makes it harder to open the github page when you try to find the download.

File: pkgs/development/tools/gomacro/default.nix [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+  vendorSha256 = "1ib4h57drikyy5aq4ms6vc1p29djlpjrh7xd3bgyykr9zmm2w1kx";
+  subPackages = [ "." ];
+
+  meta = with lib; {
+    description = "Interactive Go interpreter and debugger with generics and macros";
+    homepage = "https://github.com/cosmos72/gomacro";
+    license = licenses.mpl20;
+    maintainers = with maintainers; [ shofius ];
+    platforms = platforms.linux;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/gomacro/default.nix:22]:
```suggestion
```

Should work on darwin, too.

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (kf5grd) [reply maintainers/maintainer-list.nix:8420]: Should that be a separate commit in the same PR, or should I make a new PR for that?

Comment (kalbasit) [reply maintainers/maintainer-list.nix:8420]: Same PR, just a separate commit.

File: pkgs/development/tools/gomacro/default.nix [hunk 6]
```diff
@@ -0,0 +1,24 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
+  pname = "gomacro";
```
Comment (kalbasit) [inline pkgs/development/tools/gomacro/default.nix:4]: If you're using a commit instead of a tag, then the name must have an unstable suffix.


```suggestion
  pname = "gomacro-unstable";
```

Comment (SuperSandro2000) [reply pkgs/development/tools/gomacro/default.nix:4]: @fabaff suggested to use the release tag so this wouldn't apply then anymore. Also this goes before the version if only one variant exists.

File: pkgs/development/tools/gomacro/default.nix [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+  rev = "b4c3ab9b218fd12f22759354f4f3e37635828d1f";
+  version = "20210131-${lib.strings.substring 0 7 rev}";
```
Comment (kf5grd) [inline pkgs/development/tools/gomacro/default.nix:6]: I'm not 100% sure I did this correctly. I verified this works, but would it be preferred to do something like
```nix
let
rev = "b4c3ab9b218fd12f22759354f4f3e37635828d1f";
in
buildGoModule rec {
    pname = "gomacro";
    version = "20210131-${lib.strings.substring 0 7 rev}";
```
...etc

@fabaff @SuperSandro2000

Comment (SuperSandro2000) [reply pkgs/development/tools/gomacro/default.nix:6]: You can leave it like this. let in is not strictly required here.

**Comment by SuperSandro2000** [comment]
Can you squash the package related commits together with the commit message ``gomacro: init at 2.7``.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111405` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>gomacro</li>
</ul>
</details>

**Comment by kf5grd** [comment]
> Can you squash the package related commits together with the commit message `gomacro: init at 2.7`.

done

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:8420]: Please put this into a separate commit with the message ``maintainers: add shofius``.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (kf5grd) [reply maintainers/maintainer-list.nix:8420]: done

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111405` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>gomacro</li>
</ul>
</details>
